metriq_gym/process.py

```diff
@@ -43,12 +43,15 @@ def poll_job_results(jobs_file: str) -> list[BenchJobResult]:
             result = BenchJobResult(**(json.loads(line)))
             job = get_job(result)
             status = job.status()
-            if status == JobStatus.RUNNING:
+            if (status == JobStatus.RUNNING) or (result.job_type != job_type):
                 # Still running
                 lines_out.append(line)
             elif status == DONE:
                 # Success
-                results.append(get_job_result(job, result))
+                result.job = job
+                if job_type == BenchJobType.QV:
+                    result = get_job_result(job, result)
+                results.append(result)
             # else: # Failure
```

Comment:
Should we `raise` an error here with a message to indicate that the process failed? For instance:

```py



from qiskit.providers.exceptions import JobError

....
else:
    raise JobError(f"Job failed with status: {status}.")
```

Reply:
This would block the queue entirely at the failing job. However, we can print a message indicating this.